ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -129,6 +148,14 @@ public synchronized void stop() {
       } catch (IOException e) {
         throw new RuntimeException("Couldn't unlock rootDir: " + rootDirectory.getAbsolutePath(), e);
       }
+
+      try {
```

Comment:
You probably want to be creating the clean file while you still have the file lock to avoid issues with another cache manager starting up while you are shutting down.